### DISCUSSION TITLE
Trying to open the vault login page results in 404 not found. The server log has error `[ERROR] No matching routes for POST /cgi-bin/luci/;stok=/locale?form=lang application/x-www-form-urlencoded`

### DISCUSSION THREAD

**Discussion by sw8fbar** [dani-garcia/vaultwarden#3178] [Q&A]
### Subject of the issue
Trying to open the vault login page results in 404 not found. The server log has error `[ERROR] No matching routes for POST /cgi-bin/luci/;stok=/locale?form=lang application/x-www-form-urlencoded`

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.27.0
* Web-vault version: v2022.12.0
* Running within Docker: true (Base: Alpine)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: false (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: false
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.39.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, INVITATIONS_ALLOWED, ADMIN_TOKEN, SMTP_HOST, SMTP_SECURITY, SMTP_PORT, SMTP_FROM, SMTP_USERNAME, SMTP_PASSWORD

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": false,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://******************",
  "domain_origin": "*****://******************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "***********************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "****************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "*************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* Install method:

* Clients used: 
web vault
* Reverse proxy and version: 
Caddy
* MySQL/MariaDB or PostgreSQL version: 
SQLite
* Other relevant details:

### Steps to reproduce

docker-compose

### Expected behaviour
To see the vault login page

### Actual behaviour
Page not found!
server error log:
`[ERROR] No matching routes for POST /cgi-bin/luci/;stok=/locale?form=lang application/x-www-form-urlencoded`
### Troubleshooting data
[2023-01-27 06:18:10.290][_][ERROR] No matching routes for POST /cgi-bin/luci/;stok=/locale?form=lang application/x-www-form-urlencoded.
[2023-01-27 06:18:10.291][_][WARN] Responding with registered (not_found) 404 catcher.
[2023-01-27 06:18:10.305][_][WARN] Response was `None`.
[2023-01-27 06:18:10.305][_][WARN] Responding with registered (not_found) 404 catcher.
[2023-01-27 06:18:10.350][_][WARN] Response was `None`.
[2023-01-27 06:18:10.350][_][WARN] Responding with registered (not_found) 404 catcher.

**Comment by BlackDex** [comment]
I suggest to recheck your reverse proxy config and try again.
These kind of paths have nothing to do Bitwarden or Vaultwarden.

**Reply by sw8fbar** [reply]
I doubt that is the case as the previous version was working just fine and the configuration on the reverse proxy has not changed. Also the admin url (domain.com/admin) is working fine. This error is only on the vault page.

**Reply by BlackDex** [reply]
It really must be something on your own side. All other people have a working instance. And those paths are not coming from Bitwarden/Vaultwarden in any way. If this was a bug on the Vaultwarden side there would have been a ton of messages here regarding that.

So check the following.
- Use a different browser
- Use a different browser in Private/incognito modes
- Check the logs of your reverse proxy
- Also, check the date/time because that seems out of sync or incorrect
